### PR TITLE
[#26] Backend implementation to submit survey

### LINKED ISSUE #26: [Backend] As a user, I can submit my survey response
## Why
At the final step, users must be able to submit their responses. The mobile app must show the Submit button instead of the Next button to inform users that the app will submit their response on the next action.

## Acceptance Criteria
Call `POST /api/v1/responses` endpoint with the following body:
```
{
  "survey_id": "{{survey-id}}",
  "questions": [
    {{question-id-with-answers}},
    ...
  ]
}
```

For question-id-with-answers, there are two possible types of this data structure:
1. With ID
```
{
"id": "{{question-id}}",
"answers": [
 {
   "id": "{{answer-id}}"
 }
]
}
```

2. With ID and Answer
```
{
"id": "{{question-id}}",
"answers": [
 {
   "id": "{{answer-id}}",
   "answer": "{{answer}}"
 }
]
}
```

## Resources
[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#8de38274-3f9f-4b27-8122-58d7399b4c69)

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #26 

## What happened 👀

At the final step, users must be able to submit their responses. The mobile app must show the Submit button instead of the Next button to inform users that the app will submit their response on the next action.

## Insight 📝

- Add `POST /api/v1/responses` endpoint
- Add request object
- Add repository implementation

## Proof Of Work 📹
Screenshot_20220905_183010

Screenshot_20220905_183024

### CODE REVIEW

**Review by AVI5HEK — APPROVED** [review]
LGTM 🚀

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/network/Api.kt [hunk 1]
```diff
@@ -26,4 +27,7 @@ interface Api {
 
     @GET("api/v1/surveys/{surveyId}")
     suspend fun getSurveyDetail(@Path("surveyId") surveyId: String): BaseResponse<SurveyResponse>
+
+    @POST("api/v1/responses")
+    suspend fun submitSurvey(@Body submitSurveyRequest: SubmitSurveyRequest): BaseResponse<SubmitSurveyRequest>
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/network/Api.kt]: Is the response for this request the same as the body? Is it not just an empty response with success or failure status? 🤔

**Review by ryan-conway — COMMENTED** [review]
Just one concern, the rest LGTM 👍🏻

**Review by ryan-conway — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**